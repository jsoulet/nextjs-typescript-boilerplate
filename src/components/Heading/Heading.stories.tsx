import React from 'react'
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0'

import Heading, { HeadingProps } from './index'

export default {
  title: 'Core/Heading',
  component: Heading,
} as Meta

const Template: Story<HeadingProps> = (args) => <Heading {...args} />

export const Main = Template.bind({})
Main.args = {
  level: 1,
  children: `You may say that I'm a dreamer, but I'm not the only one`,
}

export const Heading1 = () => <Heading level={1}>Imagine there's no heaven</Heading>
export const Heading2 = () => <Heading level={2}>It's easy if you try</Heading>
export const Heading3 = () => <Heading level={3}>No hell below us</Heading>
export const Heading4 = () => <Heading level={4}>Above us only sky</Heading>
