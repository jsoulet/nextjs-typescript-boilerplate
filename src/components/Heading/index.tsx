/** @jsx jsx */
import React from 'react'
import { jsx } from '@emotion/core'
import tokens from '../tokens'
import Text from '../Text/'
export interface HeadingProps {
  level?: 1 | 2 | 3 | 4
}
type HeadingTag = 'h1' | 'h2' | 'h3' | 'h4'
const HeadingSize = {
  h1: tokens.font.size.xxl,
  h2: tokens.font.size.xl,
  h3: tokens.font.size.l,
  h4: tokens.font.size.m,
}

// const styles =css({ letterSpacing: '-1px' })

const Heading: React.FC<HeadingProps> = ({ level = 1, children, ...props }) => {
  const H = `h${level}` as HeadingTag
  return (
    <Text
      as={H}
      css={{ letterSpacing: '-1px', fontSize: HeadingSize[H], fontWeight: 'bold' }}
      {...props}
    >
      {children}
    </Text>
  )
}

export default Heading
