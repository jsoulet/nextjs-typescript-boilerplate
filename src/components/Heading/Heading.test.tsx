import { render } from '@testing-library/react'
import { Main, Heading1, Heading2, Heading3, Heading4 } from './Heading.stories'

describe('Heading', () => {
  it('matches snapshot', () => {
    const { container } = render(<Main>{`You may say that I'm a dreamer`}</Main>)
    expect(container).toMatchSnapshot()
  })
  it(`renders H1 tag`, () => {
    const { container } = render(<Heading1 />)
    const tag = container.querySelector(`h1`)
    expect(tag).toHaveTextContent(`Imagine there's no heaven`)
  })
  it(`renders H2 tag`, () => {
    const { container } = render(<Heading2 />)
    const tag = container.querySelector(`h2`)
    expect(tag).toHaveTextContent(`It's easy if you try`)
  })
  it(`renders H3 tag`, () => {
    const { container } = render(<Heading3 />)
    const tag = container.querySelector(`h3`)
    expect(tag).toHaveTextContent(`No hell below us`)
  })
  it(`renders H3 tag`, () => {
    const { container } = render(<Heading4 />)
    const tag = container.querySelector(`h4`)
    expect(tag).toHaveTextContent(`Above us only sky`)
  })
})
