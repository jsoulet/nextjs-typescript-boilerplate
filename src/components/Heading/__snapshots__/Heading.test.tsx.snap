// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading matches snapshot 1`] = `
<div>
  <h1
    class="css-1n7yzlh-Heading-Heading"
  >
    You may say that I'm a dreamer
  </h1>
</div>
`;
